crates/operator_2/src/builder.rs

```diff
@@ -0,0 +1,334 @@
+            self.operator_state_retriever,
+            self.http_rpc_url.clone(),
+        )
+        .await
+        .unwrap();
+        let is_registered = avs_registry_reader
+            .is_operator_registered(self.operator_addr)
+            .await?;
+        info!("is_operator2_registered {}", is_registered);
+        if is_registered {
+            info!("Starting operator");
+
+            let ws = WsConnect::new(self.ws_rpc_url.clone());
+            let provider = ProviderBuilder::new().on_ws(ws).await?;
+
+            let filter = Filter::new().event_signature(NewTaskCreated::SIGNATURE_HASH);
+            let sub = provider.subscribe_logs(&filter).await?;
+            let mut stream = sub.into_stream();
+
+            while let Some(log) = stream.next().await {
+                let task_option = log
+                    .log_decode::<IncredibleSquaringTaskManager::NewTaskCreated>()
+                    .ok();
+                if let Some(task) = task_option {
+                    let data = task.data();
+                    let new_task_created = NewTaskCreated {
+                        task: data.task.clone(),
+                        taskIndex: data.taskIndex,
+                    };
+                    info!(
+                        "operator2 picked up a new task , index: {} ",
+                        data.taskIndex
+                    );
+                    incredible_metrics::increment_num_tasks_received();
+                    let task_response = self.process_new_task(new_task_created);
+                    let signed_task_response = self.sign_task_response(task_response)?;
+                    if let Some(client) = &self.client {
+                        let _ = client.send_signed_task_response(signed_task_response).await;
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    /// Sign the task response
+    pub fn sign_task_response(
+        &self,
+        task_response: TaskResponse,
+    ) -> Result<SignedTaskResponse, OperatorError> {
+        let encoded_response = TaskResponse::abi_encode(&task_response);
+        let hash_msg = keccak256(encoded_response);
+
+        let signed_msg = self.key_pair.sign_message(hash_msg.as_slice());
+        let signed_task_response =
+            SignedTaskResponse::new(task_response, signed_msg, self.operator_id);
+        Ok(signed_task_response)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use alloy::primitives::Bytes;
+    use alloy::primitives::U256;
+    use ark_ec::AffineRepr;
+    use ark_ff::PrimeField;
+    use eigen_crypto_bn254::utils::verify_message;
+    use incredible_bindings::incrediblesquaringtaskmanager::IIncredibleSquaringTaskManager::Task;
+    use incredible_testing_utils::{
+        get_incredible_squaring_operator_state_retriever,
+        get_incredible_squaring_registry_coordinator,
+    };
+    use std::str::FromStr;
+    const INCREDIBLE_CONFIG_FILE: &str = r#"
```

Comment:
This config can be loaded from a text file to ease testing of different scenarios.